### DISCUSSION TITLE
What version supports Chrome 108.0.5359.125?

### DISCUSSION THREAD

**Discussion by GarretMook** [webdriverio/webdriverio#9580] [General]
In package-lock.json I see the webdriverio package is set to automatically grab the highest release with "^"

`"webdriverio": "^7.9.0"`

I want to tell it a specific version that supports Chrome version 108.0.5359.125. To avoid this error below. That way I do not have to update chrome each time the library has a new release.

```
Could not start a new session. Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 109
Current browser version is 108.0.5359.125 with binary path C:\Program Files\Google\Chrome\Application\chrome.exe
```

Would v 7.29.1 work or how can I tell?

Thanks.

**Comment by RossVertizan** [comment]
Try:
```npm install chromedriver@108```

or the Yarn equivalent if you are on Yarn.

**Reply by RossVertizan** [reply]
Sorry, that was a rather quick, terse reply.  I believe the problem you are having is not related to the version of ```webdriverio``` you have installed, but instead to the version of ```chromedriver``` you have installed which must match your version of Chrome.  Hence, the frequent need to update the version of chromedriver as shown in my response above.

Hope this helps.

**Reply by GarretMook** [reply]
Usually I do Python and just manually download the version I need of Chromedriver via https://chromedriver.chromium.org/downloads and drag it into my project directory then launch by doing 

```
s = Service('browserdriver/chromedriver.exe')

@given('chrome has successfully launched')
def launchBrowser(context):
    context.driver = webdriver.Chrome(service=s, options=options)
```

However, the the project I am on now was from a previous owner. They use **npm**. They have no reference to Chromedriver, so I figured it was being brought in by webdriverio. Is this assumption incorrect?

**Reply by RossVertizan** [reply]
In the wdio.config.js file do you have ```chromedriver``` specified in the services section?

**Reply by RossVertizan** [reply]
I think you must have chromedriver specified in the package.json file.  The error message specifically mentions the 109 version of Chromedriver.

**Reply by GarretMook** [reply]
Thanks for the replies. Yes, it is also auto-updating to the highest version of ChromeDriver when it runs. I can confirm this. However, there is no mention of ChromeDriver in the project, even when I search the repo.

I do not have a **wdio.config.js**. In **package.json** I have this:

`npm update && npm install && npm install @wdio/selenium-standalone-service`

and 

```
dependencies": {
    "codeceptjs": "^3.0.7",
    "codeceptjs-assert": "0.0.5",
    "codeceptjs-chai": "2.3.1",
    "faker": "^5.5.3",
    "typescript": "^4.3.5",
    "webdriverio": "^7.9.0"
  },
  "devDependencies": {
    "@vue/eslint-config-prettier": "^6.0.0",
    "@wdio/cli": "^7.9.0",
    "@wdio/devtools-service": "^7.9.0",
    "@wdio/selenium-standalone-service": "^7.8.0",
    "allure-commandline": "^2.13.8",
    "babel-eslint": "^10.1.0",
    "eslint": "^7.21.0",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-vue": "^7.7.0",
    "husky": "^4.3.8",
    "lint-staged": "^10.5.4",
    "object-assign": "^4.1.1",
    "prettier": "^2.2.1"
  },
```

Does that help? I am scratching me head a bit on this still...

**Comment by RossVertizan** [comment]
OK, let's get the config file stuff straight first .. ```wdio.conf.js``` is the default name of the WebdriverIO configuration file, it can be called anything but then you would need to specify it on the command line when you run webdriverio, but (as far as I am aware) you always need a  WebdriverIO config file.  I see that you are using Codecept.js, which I think is a higher level wrapper, so that may be hiding some of this from you.

The Chromedriver is actually being taken care of (I think, though I become less certain of myself as time passes) by the ```@wdio/selenium-standalone-service```, you can find documentation on that [here](https://webdriver.io/docs/selenium-standalone-service).  You will see from that documentation that this tool takes care of all necessary drivers for you, but in the wdio configuration file you can specify which versions of the drivers to use.

**Reply by GarretMook** [reply]
I think you are correct. I do have a file `codecept.conf.js` which contains stuff like 

```
  plugins: {
    wdio: { enabled: true, services: ["selenium-standalone", "devtools"] },
```

but no mention of ChromeDriver. I'll look into updating this line:

```
  "devDependencies": {
    "@wdio/selenium-standalone-service": "^7.8.0",
```

after I read more about the versioning and what not. Thanks.

**Reply by GarretMook** [reply]
@RossVertizan to follow up. Yes, I have now learned the package `@wdio/selenium-standalone-service` includes ChromeDriver as one of the drivers that it can automatically install and manage.  It says to specify the [ChromeDriver ](https://chromedriver.chromium.org/downloads) version in the chromeOptions args like so:

```
const config = {
  output: ".\\output",
  helpers: {
    WebDriver: {
      url: testUrl,
      browser: "chrome",
      desiredCapabilities: {
        chromeOptions: {
          args: [`--version=108.0.5359.71`],
        },
      },
```

No luck with this method yet to change the ChromeDriver version. Still getting same error. For educational purposes will update with a solution once I have a working one. Thanks you for the help.

**Comment by RossVertizan** [comment]
@GarretMook on this line:
```
plugins: {
    wdio: { enabled: true, services: ["selenium-standalone", "devtools"] },
```
have you tried:
```
plugins: {
    wdio: { enabled: true, 
                services: [
                      ['selenium-standalone', { drivers: { chrome: `108.0.5359.71` } }]
               , "devtools"]
             },
...
```
Hope this helps, I'm clutching at straws a bit here as I don't use Codecept.
